octoprint_Spoolman/static/js/common/formatting.js

```diff
@@ -90,3 +90,33 @@ const calculateWeight = (length, diameter, density) => {
 
     return volume * density;
 };
+
+/**
+ *  Builds a linear-gradient CSS property from the given colors and direction
+ * @param {string | undefined} direction
+ *  Direction of the gradient
+ * @param {string | undefined} colors
+ *  Comma-separated list of colors
+ * @returns string
+ */
+const calculateGradient = (direction, colors) => {
+    if (!direction || !colors) {
```

Comment:
This condition does not belong in this function, as it does not handle gradient calculation - it's an edge case handler, for situations when no color has been set at all. Doing this edge case handling here slightly obscures the branching logic.

To better convey what's going on, and to prevent multi level ternaries in the code above, I think it would be best to introduce a new function - `calculateColorCss` or something like that. In that function, there would basically be 3 branches of code:
- the "edge case" early return - if there's no data, return the dummy value
- the "single color" case, returns the singular hex value as usual
- the "multi color" case, which would call the newly introduced `calculateGradient`

Reply:
Should the `cssProperty` be calculated in this function as well, or just the `cssValue`? Instead of returning a `string` the `calculateColorCss` would return an `Object: string, string`

Reply:
That makes sense - these two values are strictly related to each other, so returning them both in one go is definitely beneficial for code clarity.